docs/core/extensions/snippets/channels/Program.Producer.cs

```diff
@@ -0,0 +1,61 @@
+﻿internal static partial class Program
+{
+    // <whiletrywrite>
+    static void ProduceWithWhileAndTryWrite(
+        ChannelWriter<Coordinates> writer, Coordinates coordinates)
+    {
+        while (coordinates is { Latitude: < 90, Longitude: < 180 })
+        {
+            var tempCoordinates = coordinates with
+            {
+                Latitude = coordinates.Latitude + .5,
+                Longitude = coordinates.Longitude + 1
+            };
+
+            if (writer.TryWrite(item: tempCoordinates))
+            {
+                coordinates = tempCoordinates;
+            }
+        }
+    }
+    // </whiletrywrite>
+
+    // <whilewrite>
+    static async ValueTask ProduceWithWhileWriteAsync(
+        ChannelWriter<Coordinates> writer, Coordinates coordinates)
+    {
+        while (coordinates is { Latitude: < 90, Longitude: < 180 })
+        {
+            await writer.WriteAsync(
+                item: coordinates = coordinates with
+                {
+                    Latitude = coordinates.Latitude + .5,
+                    Longitude = coordinates.Longitude + 1
+                });
+        }
+
+        writer.Complete();
+    }
+    // </whilewrite>
+
+    // <waittowrite>
+    static async ValueTask ProduceWithWaitToWriteAsync(
+        ChannelWriter<Coordinates> writer, Coordinates coordinates)
+    {
+        while (coordinates is { Latitude: < 90, Longitude: < 180 } &&
+            await writer.WaitToWriteAsync())
+        {
+            writer.TryWrite(
+                item: coordinates = coordinates with
+                {
+                    Latitude = coordinates.Latitude + .5,
+                    Longitude = coordinates.Longitude + 1
+                });
```

Comment:
This is different from your previous examples.  You're updating coordinates to the new value even in the case where TryWrite returns false.  So the producer may lose data.  This is essentially a manual implementation of DropWritten.

Reply:
Good catch